core/templates/dev/head/domain/objects/NumberWithUnitsObjectFactory.js

```diff
@@ -176,189 +193,188 @@ oppia.factory('NumberWithUnitsObjectFactory', [
+          var s = unit;
+          var power = 1;
         }
-      } else if (unitList[ind] === ')') {
-        var elem = parenthesisStack.pop();
-        multiplier = parseInt(elem[1]) * multiplier;
-      } else if (isunit(unitList[ind])) {
-        unitsWithMultiplier.push([unitList[ind], multiplier]);
-        // If previous element was division then we need to invert
-        // multiplier.
-        if (unitList[ind - 1] === '/') {
-          multiplier = -multiplier;
+        if (!(s in unitDict)) {
+          unitDict[s] = 0;
         }
+        unitDict[s] += multiplier * power;
       }
-    }
-    return unitsWithMultiplier;
-  };
-
-  var convertUnitDictToList = function(unitDict) {
-    var unitList = [];
-    for (var key in unitDict) {
-      unitList.push({unit: key, exponent: unitDict[key]});
-    }
-    return unitList;
-  };
-
-  var unitToList = function(unitsWithMultiplier) {
-    var unitDict = {};
-    for (var i = 0; i < unitsWithMultiplier.length; i++) {
-      var unit = unitsWithMultiplier[i][0];
-      var multiplier = unitsWithMultiplier[i][1];
-      var ind = unit.indexOf('^');
-      if (ind > -1) {
-        var s = unit.substr(0, ind);
-        var power = parseInt(unit.substr(ind + 1));
-      } else {
-        var s = unit;
-        var power = 1;
-      }
-      if (!(s in unitDict)) {
-        unitDict[s] = 0;
+      return convertUnitDictToList(unitDict);
+    };
+
+    Units.prototype.toDict = function() {
+      return {
+        units: this.units
+      };
+    };
+
+    Units.fromList = function(unitsList) {
+      return new Units(unitsList);
+    };
+
+    Units.fromStringToList = function(unitsString) {
+      return unitToList(unitWithMultiplier(Units.stringToLexical(unitsString)));
+    };
+
+    Units.prototype.toString = function() {
+      var unit = '';
+      for (var i = 0; i < this.units.length; i++) {
+        var d = this.units[i];
+        if (d.exponent === 1) {
+          unit += d.unit + ' ';
+        } else {
+          unit += d.unit + '^' + d.exponent.toString() + ' ';
+        }
       }
-      unitDict[s] += multiplier * power;
-    }
-    return convertUnitDictToList(unitDict);
-  };
-
-  Units.prototype.toDict = function() {
-    return {
-      units: this.units
+      return unit.trim();
     };
-  };
-
-  Units.fromList = function(unitsList) {
-    return new Units(unitsList);
-  };
-
-  Units.fromStringToList = function(unitsString) {
-    return unitToList(unitWithMultiplier(Units.stringToLexical(unitsString)));
-  };
-
-  Units.prototype.toString = function() {
-    var unit = '';
-    for (var i = 0; i < this.units.length; i++) {
-      var d = this.units[i];
-      if (d.exponent === 1) {
-        unit += d.unit + ' ';
-      } else {
-        unit += d.unit + '^' + d.exponent.toString() + ' ';
+
+    Units.createCurrencyUnits = function() {
+      // Creates user-defined currency (base + sub) units.
+      math.createUnit('dollar', {aliases: CURRENCY_UNITS.dollar});
```

Comment:
It would be better if we can do this, too, through currency list.

Reply:
done